JetNews/app/src/main/java/com/example/jetnews/ui/interests/InterestsScreen.kt

```diff
@@ -460,77 +416,85 @@ private fun InterestsTabRowContent(
-        PeopleList(
-            people = uiState.people,
-            selectedPeople = selectedPeople,
-            onPersonSelect = { interestsViewModel.togglePersonSelected(it) }
-        )
-    }
+        // Keep track of the height of each row to calculate the layout's final size
+        val rowHeights = IntArray(measurables.size / columns + 1)
+        // Measure elements with their maximum width and keep track of the height
+        val placeables = measurables.mapIndexed { index, measureable ->
+            val itemConstraints = if (itemWidth == Int.MAX_VALUE) {
+                outerConstraints
+            } else {
+                outerConstraints.copy(maxWidth = itemWidth)
+            }
+            val placeable = measureable.measure(itemConstraints)
+            // Update the height for each row
+            val row = index.floorDiv(columns)
+            rowHeights[row] = max(rowHeights[row], placeable.height)
+            placeable
+        }
 
-    val publicationSection = TabContent(Sections.Publications) {
-        val selectedPublications by interestsViewModel.selectedPublications.collectAsState()
-        PublicationList(
-            publications = uiState.publications,
-            selectedPublications = selectedPublications,
-            onPublicationSelect = { interestsViewModel.togglePublicationSelected(it) }
-        )
+        // Calculate maxHeight of the Interests layout. Heights of the row + top padding
+        val layoutHeight = topPaddingPx + rowHeights.sum()
+        // Calculate maxWidth of the Interests layout
+        val layoutWidth = itemWidth * columns + (itemSpacingPx * (columns - 1))
+        // Lay out given the max width and height
+        layout(
+            outerConstraints.maxWidth.coerceIn(0, layoutWidth),
+            outerConstraints.maxHeight.coerceIn(0, layoutHeight)
+        ) {
+            // Track the y co-ord we have placed children up to
+            var yPosition = topPaddingPx
+            placeables.forEachIndexed { index, placeable ->
+                // If there are multiple columns, then place items relative to each other
+                if (index % columns == 0) {
+                    // Position item on a new row
+                    placeable.placeRelative(x = 0, y = yPosition)
+                } else {
+                    // Place element in the same row, the previous item exists at this point
+                    val previousPlaceable = placeables[index - 1]
+                    placeable.placeRelative(
+                        x = previousPlaceable.width + itemSpacingPx,
+                        y = yPosition
+                    )
+                }
+                // Increment the y co-ord with the height of the row if last item of the row
+                if ((index + 1) % columns == 0) {
+                    yPosition += rowHeights[index.floorDiv(columns)]
+                }
+            }
```

Comment:
Ah, this `x` logic is also not quite general enough for more than 2 columns: We need to keep a running position for the entire row.

I think we can use `chunked` to clean this up a bit more too:
```suggestion
            placeables.chunked(columns).forEachIndexed { rowIndex, row ->
                var xPosition = 0
                row.forEach { placeable ->
                    placeable.placeRelative(x = xPosition, y = yPosition)
                    xPosition += placeable.width + itemSpacingPx
                }
                yPosition += rowHeights[rowIndex]
            }
```

Reply:
Oh wow! `chunked` is amazing! I was trying to look for an operator like it but couldn't find it. Thanks!!